src/inmanta/data/__init__.py

```diff
@@ -5153,25 +5159,24 @@ async def insert_sets_and_resources(
             )
 
     @classmethod
-    async def clear_resource_sets(
+    async def clear_resource_sets_in_version(
         cls,
         environment: uuid.UUID,
-        version: int | None = None,
+        version: int,
         *,
         connection: asyncpg.connection.Connection,
     ) -> None:
         """
-        Deletes entries on resource_set_configuration_model that relate to this environment and version (if applicable).
+        Deletes entries on resource_set_configuration_model that relate to this environment and version.
         Deletes resource sets that no longer have entries in resource_set_configuration_model
-        Deletes resources that will no longer have a valid resource set
+        Deletes resources associated with those resource sets (via cascade).
 
         :param environment: The environment from which to delete the resource sets
         :param version: The version to delete from the resource_set_configuration_model table.
-            None if we want to delete every version (i.e. when clearing the environment).
         :param connection: The connection to use
         """
-        resource_sets_to_delete = (
-            """
+
+        query = """
```

Comment:
For this query, isn't it as simple as below? You wouldn't need the `resource_sets_to_delete` at all.
```SQL
DELETE FROM resource_set AS rs
USING deleted_resource_set_versions AS del_rscm
WHERE
        rs.id = del_rscm.id
        AND NOT EXISTS (
                SELECT 1
                FROM resource_set_configuration_model AS rscm
                WHERE environment = $1 AND rscm.id = rs.id
        )
```

I may be missing something.

Reply:
I believe that is also correct.